fairmq/nanomsg/FairMQSocketNN.cxx

```diff
@@ -82,7 +82,10 @@ FairMQSocketNN::FairMQSocketNN(const string& type, const string& name, const str
 
 #ifdef NN_RCVMAXSIZE
     int rcvSize = -1;
-    nn_setsockopt(fSocket, NN_SOL_SOCKET, NN_RCVMAXSIZE, &rcvSize, sizeof(rcvSize));
+    if (nn_setsockopt(fSocket, NN_SOL_SOCKET, NN_RCVMAXSIZE, &rcvSize, sizeof(rcvSize)) != 0)
```

Comment:
It could, but I think error message is enough. The transport is still usable if this fails, with max message size of 1024kB.